R/transform-code.R

```diff
@@ -60,32 +67,39 @@ separate_chunks <- function(lines) {
   lst(r_chunks, text_chunks)
 }
 
-#' Identifies raw R code chunks
+#' Identifies raw Rmd or Rnw code chunks
 #'
 #' Raw in the sense that these chunks don't contain pure R code, but they
 #' contain a header and footer of markdown. Only code chunks that have an engine
 #' whose name matches `engine-pattern` are considered as R code.
 #' @inheritParams separate_chunks
 #' @param engine_pattern A regular expression that must match the engine name.
 #' @keywords internal
-identify_r_raw_chunks <- function(lines, engine_pattern = get_engine_pattern()) {
-  pattern <- get_knitr_pattern(lines)
+identify_raw_chunks <- function(lines, filetype, engine_pattern = get_engine_pattern()) {
+  pattern <- get_knitr_pattern(filetype)
   if (is.null(pattern$chunk.begin) || is.null(pattern$chunk.end)) {
     stop("Unrecognized chunk pattern!", call. = FALSE)
   }
-  chunks <- grep("^[\t >]*```+\\s*", lines, perl = TRUE)
-  starts <- odd(chunks)
-  ends <- even(chunks)
+
+  if (filetype == "Rmd") {
+    chunks <- grep("^[\t >]*```+\\s*", lines, perl = TRUE)
+    starts <- odd(chunks)
+    ends <- even(chunks)
+    is_r_code <- grepl(
+      paste0("^[\t >]*```+\\s*\\{\\s*", engine_pattern, "[\\s\\},]"),
+      lines[starts],
+      perl = TRUE
+    )
+  } else if (filetype == "Rnw") {
+    starts <- grep(pattern$chunk.begin, lines, perl = TRUE)
+    ends <- grep(pattern$chunk.end, lines, perl = TRUE)
+    is_r_code <- rep(TRUE, length(start))
```

Comment:
I think we are missing an s here

```suggestion
    is_r_code <- rep(TRUE, length(starts))
```

Not causing any problems because
```r
start
#> function (x, ...) 
#> UseMethod("start")
#> <bytecode: 0x107af2ed8>
#> <environment: namespace:stats>
```

Reply:
Well spotted!